src/main/scala/pipedsl/common/Syntax.scala

```diff
@@ -574,6 +576,6 @@ object Syntax {
   //This is an already "locked" memory (i.e. one line instantiation, no reference to the unlocked memory)
   case class CirLockMem(elemTyp: Type, addrSize: Int, impl: LockInterface, szParams: List[Int], numPorts: Int) extends CirExpr
   case class CirLockRegFile(elemTyp: Type, addrSize: Int, impl: LockInterface, szParams: List[Int]) extends CirExpr
-  case class CirNew(mod: Id, mods: List[Id], params: List[EInt]) extends CirExpr
+  case class CirNew(mod: Id, specialized :List[Int], mods: List[Id], params: List[EInt]) extends CirExpr
```

Comment:
What is this name, `specialized` ? I'm not sure what it's meant to capture.

Reply:
the ints which specialize the generics of mod. I have added a comment explaining this, so resolving.